apisix/plugins/request-id.lua

```diff
@@ -49,7 +49,7 @@ function _M.rewrite(conf, ctx)
     end
 
     if conf.include_in_response then
-        ctx.x_request_id = uuid_val
+        ctx[conf.header_name] = uuid_val
```

Comment:
I think we need to add a prefix like `request-id` as `conf.header_name` may override other value by accident.

Reply:
Agree +1

Reply:
I don’t really understand the intention of adding a prefix, because header_name is originally defined by the user. He should know what the definition of header_name looks like.

Reply:
But he doesn't know all the fields in the ctx and the field will be added to ctx in the future.

Reply:
Do you mean like this？
```
if conf.include_in_response then
        ctx["request-id" .. conf.header_name] = uuid_val
end
```

Reply:
`ctx["request-id-" .. conf.header_name] = uuid_val` would be better.

Reply:
All right！

In addition, I found that unit testing is not easy to do. Is there any related documentation? Thank you

Reply:
You can configure global rule and route like this one, with `/echo` uri: https://github.com/apache/apisix/blob/b76f20eddfdbd0bd546afbc515c5948a29e86600/t/node/global-rule.t#L312
https://github.com/apache/apisix/blob/b76f20eddfdbd0bd546afbc515c5948a29e86600/t/lib/server.lua#L394

And then hit it and check the header sent to upstream via :
https://github.com/apache/apisix/blob/b76f20eddfdbd0bd546afbc515c5948a29e86600/t/plugin/zipkin.t#L458-L462